test/integration/consul-container/libs/cluster/container.go

```diff
@@ -501,7 +505,7 @@ type containerOpts struct {
 	addtionalNetworks []string
 }
 
-func newContainerRequest(config Config, opts containerOpts) (podRequest, consulRequest testcontainers.ContainerRequest) {
+func newContainerRequest(config Config, opts containerOpts, ports ...int) (podRequest, consulRequest testcontainers.ContainerRequest) {
```

Comment:
I think we can per-allocated the exposed port at https://github.com/hashicorp/consul/blob/7c9801ecf6f9e1a2e9f71ddc8b655a7fd6704fc2/test/integration/consul-container/libs/cluster/container.go#L512-L526

Therefore, we don't need to change the function and group all used ports in one place

Reply:
Just as an ergonomics thing, having to explicitly add in punched ports to a global map in a separate subpackage makes things quite a bit harder to track down then to have a test specify the ports it wants punched and just do something like panic if the ports have already been allocated. Especially if you're doing some more complex setups that might involve something like 3-4 listeners on a gateway, and doing that on multiple tests this quickly grows into having to explicitly add a big chunk of forwarded, non-overlapping ports to this list.